### ISSUE TITLE
render mesh at training views

### ISSUE THREAD

**Issue by leixiao-ubc** [autonomousvision/sdfstudio#100]
Thanks for your awesome work here! I just have questions about the mesh rendering at training views.

(1) Following the instruction, I run the script below to hopefully render the mesh at trajectory interpolated from the training views. 

`ns-render-mesh --meshfile meshes/neus-facto-replica-room0.ply --traj interpolate  --output-path renders/neus-facto-replica-room0.mp4 sdfstudio-data --data data/sdfstudio-demo-data/replica-room0`

For example, I expect the first view to match the first frame in the training data:

 ("data/sdfstudio-demo-data/replica-room0/000000_rgb.png")
![000000_rgb](https://user-images.githubusercontent.com/19320556/235283363-1bff5df5-a7a2-4d62-9f72-766206124500.png)

However, the rendered views stay at the following view, which I guess is the default view of the open3d visualizer? I wonder if I misunderstand anything or if the open3d viewer didn't work with provided camera poses?

![Untitled](https://user-images.githubusercontent.com/19320556/235283297-476c6c70-9694-42d4-ad2f-df059da31986.png)

(2) Another thing I did is to take the extracted mesh, and render it with PyTorch3D. As I understand, the sdfstudio camera data e.g. "data/sdfstudio-demo-data/replica-room0/meta_data.json" follows the OpenCV coordinates. So to render the mesh in Pytorch3D, I use the function `pytorch3d.utils.cameras_from_opencv_projection()` to convert the camera parameters ("camtoworld") to Pytorch3D coordinates. However, I found that the rendered views in PyTorch3D don't match the training views e.g. "data/sdfstudio-demo-data/replica-room0/000000_rgb.png". 

To further debug this, I created a synthetic toy scene with OpenCV coordinates and used the above process to render it in PyTorch3D, and the views match well. That made me wonder if I misunderstand anything about the provided "data/sdfstudio-demo-data/replica-room0/meta_data.json" or else. 

Thanks very much for your help in advance!

**Comment by niujinshuchong** [comment]
@leixiao-ubc 

1. That's strange since we update the camera pose of open3d every frame. Please check it here: https://github.com/autonomousvision/sdfstudio/blob/master/scripts/render_mesh.py#L171 . But it seems like you are using a Windows system, which we didn't test, so I am unsure if we need to set something differently.
2. Yes, the camera poses in meta_data.json is in OpenCV format. Do you use the `--auto-orient True` in your training command?

**Comment by leixiao-ubc** [comment]
@niujinshuchong 
Thanks for your quick reply. 

1) Yes I run it on Windows. I will further test it. 

2) No I didn't set --auto-orient True but instead used the provided command `ns-train neus-facto --pipeline.model.sdf-field.inside-outside True --pipeline.model.mono-depth-loss-mult 0.1 --pipeline.model.mono-normal-loss-mult 0.05 --vis viewer --experiment-name neus-facto-replica1 sdfstudio-data --data data/sdfstudio-demo-data/replica-room0 --include_mono_prior True`.

**Comment by leixiao-ubc** [comment]
Bypassed the problem by using the original DTU images and cameras.npz.

**Comment by rockywind** [comment]
@leixiao-ubc 
I met the same  issue with you.
How can you solve it?

**Comment by niujinshuchong** [comment]
@leixiao-ubc Could you share your solution? Thanks.

**Comment by zoezhou1999** [comment]
Hi @niujinshuchong  @leixiao-ubc 
I am trying to render mesh exported from bakedsdf of sdfstudio using Pytorch3D with mipnerf-360 transform.json. if i understand correctly, sdfstudio uses OpenGL convention, so I use FoVPerspectiveCameras from pytorch3D. But no success. By the way, if i use nvdiffrast that uses OpenGL convention, I can successfully render correct image. Any help will be appreciated! Thank you!
